src/compiler/checker.ts

```diff
@@ -8540,6 +8551,13 @@ namespace ts {
             return undefined;
         }
 
+        function isIife(func: FunctionExpression | MethodDeclaration) {
+            return (func.kind === SyntaxKind.FunctionExpression || func.kind === SyntaxKind.ArrowFunction) &&
+                func.parent.kind === SyntaxKind.ParenthesizedExpression &&

```

Comment:
I suggest using the `skipParenthesizedNodes` function before checking for a call expression at the top.